packages/nuxt/src/pages/runtime/router.options.ts

```diff
@@ -28,26 +28,27 @@ export default <RouterConfig> {
     }
 
     // Hash routes on the same page, no page hook is fired so resolve here
-    if (to.path === from.path) {
-      if (from.hash && !to.hash) {
+    if (to.path === from.path && !to.hash) {
```

Comment:
should we check like this?

```suggestion
    if (to.path === from.path && (!to.hash || (to.hash !== from.hash))) {
```

Reply:
Wanted to list each case and use boolean expression simplifier, which might help decide:
- `from.hash && !to.hash` --> `{ left: 0, top: 0 }`
- `!from.hash && to.hash`  --> `{ el: to.hash, top: _getHashElementScrollMarginTop(to.hash), behavior }`
- `!from.hash && !to.hash`  --> `false` (no change)
- `from.hash && to.hash` --> `false` (no change)

`a` - `from.hash`
`b` - `to.hash`

`(a && !b) || (!a && !b) || (a && b)` => `a || !b` => `from.hash || !to.hash`
```suggestion
    if (to.path === from.path && (!to.hash || from.hash)) {
```

Reply:
Hello, don't want to rush you, but do we have any eta for this? I have used hash based navigation in a few places in my project.